src/coreclr/jit/loopcloning.cpp

```diff
@@ -853,11 +853,27 @@ BasicBlock* LoopCloneContext::CondToStmtInBlock(Compiler*
     noway_assert(conds.Size() > 0);
     assert(slowPreheader != nullptr);
 
+    // For now assume high likelihood for the fast path,
+    // uniformly spread across the gating branches.
+    //
+    // For "normal" cloning this is probably ok. For GDV cloning this
+    // may be inaccurate. We should key off the type test likelihood(s).
+    //
+    // TODO: this is a bit of out sync with what we do for block weights.
+    // Reconcile.
+    //
+    const weight_t fastLikelihood = 0.999;
```

Comment:
Out of curiosity, where does this number of significant figures come from?

Reply:
It's just an arbitrary factor. For most cloned loops we actually never expect the cold loop to run, since we think it's execution may cause exceptions. But setting likelihood to zero seemed to drastic.

Reply:
`optCloneLoop` scales the fast loop blocks by `0.99`. There is a comment about it:
https://github.com/dotnet/runtime/blob/5f52977c066a29d124694eb423eff0b866df37a7/src/coreclr/jit/loopcloning.cpp#L1944-L1949

It seems like we should use the same likelihood in both places.

Reply:
Ah, good point. I will tweak this in the next round of changes.